src/IDCodeValidator.java

```diff
@@ -0,0 +1,175 @@
+
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+
+public class IDCodeValidator {
+
+
+    public static boolean isCorrect(String idCode) {
+        System.out.println(idCode);
+        IDNumberData personalCode = IDNumberData.getInformationFromIdCode(idCode);
+        if (idCode.length() == 11 && isGenderNumberCorrect(personalCode.genderNumber) && isDayNumberCorrect(personalCode.dayNumber) && isMonthNumberCorrect(personalCode.monthNumber) && isBirthDateCorrect(personalCode.yearNumber, personalCode.monthNumber, personalCode.dayNumber) && checkControlNumber(idCode)) {
+            System.out.println("ID is correct");
+            return true;
+        }
+        if (idCode.length() != 11) {
+            System.out.println("The length is not correct");
+
+        }
+
+        return false;
+    }
+
+
+    public static boolean isGenderNumberCorrect(int genderNumber) {
+        if (genderNumber == 1 || genderNumber == 3 || genderNumber ==5){
+            return true;
+        }
+        if (genderNumber == 2 || genderNumber == 4 || genderNumber ==6) {
+            return true;
+        }
+        return false;
+    }
+
+
+    public static boolean isDayNumberCorrect(int dayNumber) {
+        if (dayNumber > 0 && dayNumber <= 31) {
+            System.out.println(dayNumber);
+            return true;
+        }
+        System.out.println("There is no such day");
+        return false;
+    }
+
+    public static boolean isMonthNumberCorrect(int monthNumber) {
+        if (monthNumber > 0 && monthNumber <= 12) {
+            System.out.println(monthNumber);
+            return true;
+        }
+        System.out.println("There is no such month");
+        return false;
+    }
+
+
+    public static boolean isBirthDateCorrect(int yearNumber, int monthNumber, int dayNumber) {
+        LocalDateTime birthday = LocalDateTime.of(1998, 01, 29, 0, 0);
+        LocalDateTime now = LocalDateTime.now();
+        long yearsOld = ChronoUnit.YEARS.between(birthday, now);
+        System.out.println(yearsOld);
+
+        if (monthNumber == 2 && isLeapYear(yearNumber) && yearsOld > 0) {
+            System.out.println(yearsOld + 1);
+            return true;
+        }
+        if ((!(yearsOld > 0))) {
+            System.out.println("Error");
+        }
+
+        return false;
+    }
+
+
+    public static boolean isLeapYear(int yearNumber) {
+        if ((yearNumber % 4 == 0 && yearNumber % 100 != 0) || (yearNumber % 400 == 0)) {
+            return true;
+        } else return false;
```

Comment:
Please please please, 
use the `{}` all the time, even though Java allows to do `else return false` it makes reading code harder
This is more readable
```        
if ((yearNumber % 4 == 0 && yearNumber % 100 != 0) || (yearNumber % 400 == 0)) {
            return true;
        } else {
            return false;
        }
```